Tests/HarmonyTests/PutNetworkDataSourceTests.swift

```diff
@@ -0,0 +1,394 @@
+        let response = Utils.provideResponse(url: url, statusCode: 200, httpVersion: "HTTP/2.0", headers: ["json": "application/json; charset=utf-8"])
+        let dataSource: PutNetworkDataSource<Entity> = providePutDataSource(url: url, request: request, response: response)
+        
+        // Then
+        expectPutError(dataSource, query, CoreError.DataSerialization(), .putAll)
+    }
+
+    func test_putall_returns_value_when_provided_as_parameter() {
+        // Given
+        let array = [Entity(name: "", owner: "")]
+        let query = NetworkQuery(method: .put(type: nil), path: "")
+        let url = "www.dummy.com"
+        let request = Utils.provideRequest(url: url, cachePolicy: .reloadIgnoringLocalAndRemoteCacheData, timeout: 1.0)
+        let response = Utils.provideResponse(url: url, statusCode: 200, httpVersion: "HTTP/2.0", headers: ["json": "application/json; charset=utf-8"])
+        let dataSource: PutNetworkDataSource<Entity> = providePutDataSource(url: url, request: request, response: response, jsonFileName: "EntityList")
+        
+        // Then
+        expectPutAll(value: array, dataSource, query, nil)
+    }
+
+    func test_putall_returns_data_serialization_error_when_data_fails_to_be_parsed_as_generic_type() {
+        // Given
+        let query = NetworkQuery(method: .put(type: nil), path: "")
+        let url = "www.dummy.com"
+        let decoder = DecoderSpy()
+        decoder.forceFailure = true
+        let request = Utils.provideRequest(url: url, cachePolicy: .reloadIgnoringLocalAndRemoteCacheData, timeout: 1.0)
+        let response = Utils.provideResponse(url: url, statusCode: 200, httpVersion: "HTTP/2.0", headers: ["json": "application/json; charset=utf-8"])
+        let dataSource: PutNetworkDataSource<Entity> = providePutDataSource(url: url, request: request, response: response, decoder: decoder)
+        
+        // Then
+        expectPutAll(value: [Entity(name: "", owner: "")], dataSource, query, CoreError.DataSerialization())
+    }
+    
+    // Validation
+    func test_put_returns_query_not_supported_error_when_query_not_a_network_query() {
+        // Given
+        let dataSource: PutNetworkDataSource<Int> = providePutDataSource(url: "")
+        let query = AllObjectsQuery()
+        
+        // Then
+        expectPutError(dataSource, query, CoreError.QueryNotSupported(), .put)
+    }
+
+    func test_put_returns_query_not_supported_error_when_query_method_not_put() {
+        // Given
+        let dataSource: PutNetworkDataSource<Int> = providePutDataSource(url: "")
+        let query = NetworkQuery(method: .get, path: "")
+        
+        // Then
+        expectPutError(dataSource, query, CoreError.QueryNotSupported(), .put)
+    }
+
+    func test_put_returns_illegal_argument_error_when_value_and_content_type_are_both_present() {
+        // Given
+        let dataSource: PutNetworkDataSource<Entity> = providePutDataSource(url: "")
+        let entity = Entity(name: "", owner: "")
```

Comment:
Create `anyEntity()` function with random strings for `name` and `owner`

Reply:
done